webapi/src/test/scala/org/knora/webapi/e2e/v2/StandoffRouteV2R2RSpec.scala

```diff
@@ -0,0 +1,151 @@
+
+
+/**
+  * End-to-end test specification for the search endpoint. This specification uses the Spray Testkit as documented
+  * here: http://spray.io/documentation/1.2.2/spray-testkit/
+  */
+class StandoffRouteV2R2RSpec extends R2RSpec {
+
+    override def testConfigSource: String =
+        """
+          |# akka.loglevel = "DEBUG"
+          |# akka.stdout-loglevel = "DEBUG"
+        """.stripMargin
+
+    private val responderManager = system.actorOf(Props(new ResponderManager with LiveActorMaker), name = RESPONDER_MANAGER_ACTOR_NAME)
+    private val storeManager = system.actorOf(Props(new StoreManager with LiveActorMaker), name = STORE_MANAGER_ACTOR_NAME)
+
+    private val standoffPath = StandoffRouteV2.knoraApiPath(system, settings, log)
+
+    implicit private val timeout: Timeout = settings.defaultRestoreTimeout
+
+    implicit def default(implicit system: ActorSystem): RouteTestTimeout = RouteTestTimeout(new DurationInt(15).second)
+
+    implicit val ec: ExecutionContextExecutor = system.dispatcher
+
+    private val anythingUser = SharedTestDataADM.anythingUser1
+    private val anythingUserEmail = anythingUser.email
+
+    private val password = "test"
+
+    object RequestParams {
+
+        val pathToLetterMapping = "_test_data/test_route/texts/mappingForLetter.xml"
+
+        val pathToLetterXML = "_test_data/test_route/texts/letter.xml"
+
+        val pathToLetter2XML = "_test_data/test_route/texts/letter2.xml"
+
+        val pathToLetter3XML = "_test_data/test_route/texts/letter3.xml"
+
+        // Standard HTML is the html code that can be translated into Standoff markup with the OntologyConstants.KnoraBase.StandardMapping
+        val pathToStandardHTML = "_test_data/test_route/texts/StandardHTML.xml"
+
+        val pathToHTMLMapping = "_test_data/test_route/texts/mappingForHTML.xml"
+
+        val pathToHTML = "_test_data/test_route/texts/HTML.xml"
+
+    }
+
+    private val rdfDataObjects = List(
+
+        RdfDataObject(path = "_test_data/all_data/incunabula-data.ttl", name = "http://www.knora.org/data/incunabula"),
+        RdfDataObject(path = "_test_data/demo_data/images-demo-data.ttl", name = "http://www.knora.org/data/00FF/images"),
+        RdfDataObject(path = "_test_data/all_data/anything-data.ttl", name = "http://www.knora.org/data/anything")
+
+    )
+
+    "Load test data" in {
+        Await.result(storeManager ? ResetTriplestoreContent(rdfDataObjects), 360.seconds)
+        Await.result(responderManager ? LoadOntologiesRequestV2(KnoraSystemInstances.Users.SystemUser), 30.seconds)
+    }
+
+    "The Standoff v2 Endpoint" should {
+
+        "create a mapping from a XML" in {
+
+            val xmlFileToSend = new File(RequestParams.pathToLetterMapping)
+
+            val mappingParams =
+                s"""
+                   |{
+                   |    "knora-api:mappingHasName": "LetterMapping",
+                   |    "knora-api:attachedToProject": "$ANYTHING_PROJECT_IRI",
```

Comment:
@benjamingeer I noticed that you used `knora-api:projectIri` (complex schema) in the ontology route. 

Shall I also use `knora-api:projectIri` (complex schema)?

```
"knora-api:projectIri" : {
        "@id" : "knora-api:projectIri",
        "@type" : "owl:DatatypeProperty",
        "knora-api:objectType" : "http://www.w3.org/2001/XMLSchema#anyURI",
        "rdfs:comment" : "Represents the IRI of a Knora project",
        "rdfs:label" : "project IRI"
      }
``` 

```
"knora-api:attachedToProject" : {
        "@id" : "knora-api:attachedToProject",
        "@type" : "owl:ObjectProperty",
        "knora-api:objectType" : "http://api.knora.org/ontology/knora-api/v2#knoraProject",
        "rdfs:comment" : "Connects a Resource or a ListNode (only the root node) to a project"
      }
```

Neither an ontology nor a mapping are resources. However, I think that `attachedToProject` expresses the relation between an ontology or a mapping and a project better than `projectIri`.

Reply:
You're right, I should have used `attachedToProject`. I'll change it in the ontology route.

Reply:
Then we should change the `rdfs:comment` of `attachedToProject` in `knora-admin.ttl` too. It does not have a `subjectClassConstraint` so it should be fine.

Reply:
I'll do that, too. It's already used with ontologies when they're served, so it's definitely fine.

Reply:
And it's already attached to ontologies in the triplestore.

Reply:
Ok, very good. Thank you!